cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -1302,3 +1333,46 @@ with the metadata returned as a JSON string in the return body::
      - The properties requested were returned as a JSON string in the body of the response
    * - ``404 NOT FOUND``
      - The entity, program, or run for which properties are being requested was not found
+
+**Note**: The Metadata aggregation of program, application, stream and dataset associated with a program is
+deprecated. Program runs are treated like any other entity to which metadata can be associated directly.
+To retrieve the direct metadata of program run please refer to
+:ref:`Metadata Properties <http-restful-api-metadata-properties>` and
+:ref:`Metadata Tags <http-restful-api-metadata-tags>`. An additional query parameter ``aggregateRun`` must be set to
+``false`` while operating with metadata directly associated with program runs.
+
+.. _metadata_custom_entities:
+
+Metadata for Custom Entities
+============================
+
+Metadata can also be associated with custom entities. In CDAP Entities are separated into two kinds:
+
+- CDAP Entities: These are system defined entities that have special meaning in CDAP.
+``Namespace``,``Stream``, ``Application``, ``Dataset`` etc. are example of CDAP Entities.
+
+- Custom Entities: These are user defined entities that represent a resource that exists in CDAP.
+Custom Entities are represented as a hierarchical key-value pair and can optionally have a explicitly defined type.
+
+If a type is not specified then the last key in the hierarchy is considered as the type.
+
+In REST APIs, custom entities are represented as hierarchical key-value pairs. If the last key in the hierarchy is
+not a type, the type is specified as a query parameter.
+
+For example, to add tags to a custom file entity in a dataset::
+
+  POST /v3/namespaces/<namespace-id>/datasets/<dataset-id>/file/<file-name>/metadata/tags
+
+In the example above, the custom entity is a single key-value pair where file is the key and <file-name> is the value.
```

Comment:
enclose file and <file-name> in backticks to make them stand out a bit more and indicate they are values

Reply:
Done.